### ISSUE TITLE
Null Fields Fail Length Validation

### ISSUE THREAD

**Issue by cmanallen** [marshmallow-code/marshmallow#209]
I'm using version 1.2.2.  Consider the following code:

```
favorite_band = String(validate=Length(min=1, max=50))
```

_Problem_
Submitting `{"favorite_band": ""}` fails as expected.  However, `{"favorite_band": null}` fails as well due to the length validation.  If a user wants to specify their favorite band then it should be specified as a non-zero length string.  If a user doesn't want to specify their favorite band it should be stored as `None`.

_Solution_
Length validation should not be considered when evaluating `null` values.  Length validation should only be considered when the value is a `str` type.

Basically:
Disallow this `{'favorite_band': ''}`
Allow this `{'favorite_band': null}`

**Comment by philtay** [comment]
Version 2.0 introduces the `allow_none` parameter.
